### ISSUE TITLE
[bug] Occasional crash during `iterateAncestors`

### ISSUE THREAD

**Issue by tsmethurst** [superseriousbusiness/gotosocial#850]
Here's some logs! found (and suffered) by @NyaaaWhatsUpDoc 

```
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
2902ea70 sp=0xc02902e750 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
2902ed90 sp=0xc02902ea70 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
2902f0b0 sp=0xc02902ed90 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
2902f3d0 sp=0xc02902f0b0 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
2902f6f0 sp=0xc02902f3d0 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
2902fa10 sp=0xc02902f6f0 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
2902fd30 sp=0xc02902fa10 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
29030050 sp=0xc02902fd30 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
29030370 sp=0xc029030050 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
29030690 sp=0xc029030370 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
290309b0 sp=0xc029030690 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
29030cd0 sp=0xc0290309b0 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
29030ff0 sp=0xc029030cd0 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
29031310 sp=0xc029030ff0 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
29031630 sp=0xc029031310 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
29031950 sp=0xc029031630 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
29031c70 sp=0xc029031950 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
29031f90 sp=0xc029031c70 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
290322b0 sp=0xc029031f90 pc=0x1d3e545
github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing.(*deref).iterateAncestors(0xc00047a3
f0, {0x26f83e0, 0xc0000928a0}, {0xc000778bd0, 0x3}, {{0xc0016a4440, 0x5}, {0x0, 0x0}, 0x0, ...})
        github.com/superseriousbusiness/gotosocial/internal/federation/dereferencing/thread.go:106 +0x565 fp=0xc0
290325d0 sp=0xc0290322b0 pc=0x1d3e545
```

**Comment by NyaaaWhatsUpDoc** [comment]
thank you for writing this up for my lazy ass :p

**Comment by NyaaaWhatsUpDoc** [comment]
We love a good stack overflow

**Comment by NyaaaWhatsUpDoc** [comment]
Fixed by #851
